### DISCUSSION TITLE
Refresh token called empty with NextJs

### DISCUSSION THREAD

**Discussion by MrCasquette** [directus/directus#17012] [Archive]
I have this problem for many month and never find a way to get a clean authentication. Refresh always comes to mess everything up. Sometimes i'am disconnected, sometimes when i logout it say refresh token is required. And then I thought about something, hey is this problem caused by react 18 or React Strict mode? Or is it something only with next. If someone has the same problem please help me with this to get a clean auth process with directus and nextJs :)

I will really thank you for the help

**Comment by paescuj** [comment]
See https://github.com/directus/sdk/issues/96

**Reply by MrCasquette** [reply]
I had no notification about it Thank you, I'm happy to see i'm not alone ^^

**Reply by paescuj** [reply]
You're welcome 😃

**Comment by Nisthar** [comment]
> See https://github.com/directus/sdk/issues/96

The page is not found. What is it about?

**Reply by paescuj** [reply]
The SDK has been completly rewritten a few months ago and moved into the main repo: https://github.com/directus/directus/tree/main/sdk

See also https://docs.directus.io/guides/sdk/getting-started.html.

I'd have guessed that the original issue has been solved with the new implementation.
